app/services/onestop_id.rb

```diff
@@ -182,16 +208,10 @@ def validate_hash(value)
     end
   end
 
-  LOOKUP = Hash[OnestopId::OnestopIdBase.descendants.map { |c| [[c::PREFIX, c::NUM_COMPONENTS], c] }]
   LOOKUP_MODEL = Hash[OnestopId::OnestopIdBase.descendants.map { |c| [c::MODEL, c] }]
 
   def self.handler_by_string(string: nil)
-    if string && string.length > 0
-      split = string.split(COMPONENT_SEPARATOR)
-      prefix = split[0].to_sym
-      num_components = split.size
-      LOOKUP[[prefix, num_components]]
-    end
+    OnestopId::OnestopIdBase.descendants.select { |cls| cls.match?(string) }.first
```

Comment:
Any idea on the performance implications of this change? I imagine it's called many times.